lib/puma/metrics/parser.rb

```diff
@@ -0,0 +1,67 @@
+require 'prometheus/client'
+
+module Puma
+  module Metrics
+    class Parser
+      PUMA_METRICS = %i[
+        backlog
+        booted_workers
+        max_threads
+        old_workers
+        pool_capacity
+        running
+        workers
+      ].freeze
+
+      def initialize(clustered = false)
+        register_default_metrics
+        register_clustered_metrics if clustered
+      end
+
+      def parse(metrics)
+        metrics = metrics.each_with_object({}) { |(k, v), memo| memo[k.to_sym] = v; } # symbolize
+        labels = metrics.key?(:index) ? { index: metrics[:index] } : {} # include :index if present
+        metrics.collect do |key, value|
+          send(key, labels, value) # metric_name, labels, value
```

Comment:
`public_send`? Or can you avoid `send` completely?

Reply:
good point, yes I think I can avoid `send` completely...